rfcs/0145-workerpoolid-taskqueueid.md

```diff
@@ -0,0 +1,84 @@
+# RFC 0145 - Worker Pools and Task Queues
+* Comments: [#0145](https://api.github.com/repos/taskcluster/taskcluster-rfcs/pulls/145)
+* Proposed by: @djmitche
+
+# Summary
+
+We will replace the two-level identifier `<provisionerId>/<workerType>` with a single `<workerPoolId>` (when concerning pools of workers) and `<taskQueueId>` (when concering queues of tasks).
+This will be done in such a way as to maintain backward compatibility for a suitable duration.
+
+## Motivation
+
+We have long had a "two-level" identifier space for worker types: `<provisionerId>/<workerType>`.
+With the advent of worker-manager, there's no longer any significance to these levels, and the name `provisionerId` no longer has meaning as there are no longer any "provisioner" objects to be identified.
+
+A few other aspects of this terminology have caused some confusion.
+First, we have used "worker type" to mean several things:
+ * The configuration for the pool in the AWS provisioner (e.g., `https://aws-provisioner.taskcluster.net/v1/worker-type/:worker-type`),
+ * the pool of workers of that type (e.g., `https://queue.taskcluster.net/v1/provisioners/:provisionerId/worker-types/:workerType`), and
+ * the queue containing tasks for that pool (e.g., `https://queue.taskcluster.net/v1/pending/:provisionerId/:workerType`).
+
+We have also used the field name `workerType` to refer to a worker type, which differs from the rest of Taskcluster where identifier names end in `Id`.
+
+This proposal outlines a low-impact way to address these sources of confusion as we transition to using worker-manager.
+
+# Details
+
+All worker-manager and related APIs will take a `workerPoolId` parameter to identify a pool of workers and the configuration and status of that pool.
+
+All queue and task-related APIs will take a `taskQueueId` parameter to identify the queue containing the task.
+
+Workers in a worker pool with identifier `workerPoolId` always and only claim tasks from a queue with `taskQueueId == workerPoolId`.
+Thus the two concepts describe separate, but closely linked, concepts.
```

Comment:
Interesting. If we revisit this requirement, it also opens the door for [multiple queues per workerType](https://github.com/mozilla-releng/scriptworker/issues/44#issuecomment-272526544) down the road if that becomes something we want. Alternately we could add support in some worker implementations for belonging to multiple `workerPoolId`s, again if and only if that's something we deem important enough to support.

This is tangential to this PR, but these two lines made me think of it.